packages/wp-story-editor/src/components/mediaUpload/mediaPicker/useMediaPicker.js

```diff
@@ -75,6 +84,44 @@ function useMediaPicker({
   } = useConfig();
   const { showSnackbar } = useSnackbar();
 
+  const { deleteElementsByResourceId } = useStory((state) => ({
+    deleteElementsByResourceId: state.actions.deleteElementsByResourceId,
+  }));
+
+  const { prependMedia, deleteMediaElement } = useLocalMedia(
+    ({ actions: { prependMedia, deleteMediaElement } }) => {
+      return {
+        prependMedia,
+        deleteMediaElement,
+      };
+    }
+  );
+
+  const uploadMedia = useCallback(
+    (resource) => {
+      if (['image', 'video', 'gif'].includes(resource.type)) {
+        prependMedia({
+          media: [resource],
+        });
+      }
+      if (onUploadMedia) {
+        onUploadMedia(resource);
+      }
+    },
+    [onUploadMedia, prependMedia]
+  );
+
+  const deleteMedia = useCallback(
+    (resource) => {
+      deleteMediaElement({ id: resource.id });
+      deleteElementsByResourceId({ id: resource.id });
+      if (onDeleteMedia) {
+        onDeleteMedia(resource);
+      }
+    },
+    [deleteElementsByResourceId, deleteMediaElement, onDeleteMedia]
+  );
```

Comment:
This should not be in the wp-story-editor package, since actions like prepending media and deleting media elements is something that also needs to happen in a decoupled/standalone context.